canonical-websites/snapcraft.io
# Updated channel map

# Done

 Fixes https://github.com/canonical-websites/snapcraft.io/issues/731

- Refactored `channelMap.js` to be a class
- Refactored `_channel_map.html`
- Added extra styles to `_snapcraft_details_heading.scss`
- Included `vf-u-hide`
- Removed `Size` and `Version` but added `Last updated` in the details table on the snap details page
- Better implemented `defaultTrack` throughought (hardcoded to 'node' for now, until implemented store-side)
- Made `copySuccess` a method on the global `snapcraft` so as to initialize within JS
- Added `get_lowest_available_risk` to find the lowest available risk for the default track

# QA

- Pull the branch
- `./run`
- Visit http://0.0.0.0:8004/store
- Click on some snaps, search for snaps and click on them
- For each snap click the 'Install' button, make sure the overlay works and everything within it does
- For each snap click the 'Other versions' button, make sure the overlay works and everything in does
- The default channel should always be at the top of the table
- The default channel should always have a bold name
- The 'Overview' tab (if present) should show all available risks within 'latest' and the lowest available risk for other tracks
- The 'All releases' tab (if present) should show all available tracks
